### PR TITLE
Translib phase I changes

### PR DESCRIPTION
**Pull request by anand-kumar-subramanian**
- Go formatting changes
- Subscribe API standardized to take structures as input and output (This is a dependency for REST and gNMI PRs)
- Action(RPC ) API body (Implementation will be done in future PRs (This is a dependency for REST and gNMI PRs)
- REST query parameters field support in translib (This is a dependency for REST)
- Changes to have write disabled for DB operation (Need to integrate with DB access layer once that PR comes in)
- Versioning bare minimum actual support will come in a separate PR (This is a dependency for REST and gNMI PRs)
- Static analysis tool fixes

### CODE REVIEW

File: translib/app_interface.go [hunk 1]
```diff
@@ -137,12 +148,9 @@ func getAppModuleInfo(path string) (*appInfo, error) {
...
-
-	err = errors.New(errStr)
-	log.Error(errStr)
-
-	var app *appInfo
+	/* If no specific app registered fallback to default/common app */
+	log.Infof("No app module registered for path %s hence fallback to default/common app", path)
+	app := appMap["*"]
 
 	return app, err
```
[the remark this answers is not in the document]
Comment (anand-kumar-subramanian) [reply translib/app_interface.go:155]: If it is not set it is nil by default in golang which indicates it is success.

File: translib/app_interface.go [hunk 2]
```diff
@@ -108,7 +119,7 @@ func addModel(model *ModelData) error {
 }
 
 //App modules can use this function to unregister itself from the app interface
-func unregister(path string) error {
+/*func unregister(path string) error {
```
[the remark this answers is not in the document]
Comment (anand-kumar-subramanian) [reply translib/app_interface.go]: Removed

File: translib/translib.go [hunk 3]
```diff
@@ -34,114 +34,159 @@ This package can also talk to non-DB clients.
...
+	log "github.com/golang/glog"
 )
 
 //Write lock for all write operations to be synchronized
 var writeMutex = &sync.Mutex{}
 
 //minimum global interval for subscribe in secs
 var minSubsInterval = 20
-var maxSubsInterval = 60
+var maxSubsInterval = 600
```
[the remark this answers is not in the document]
Comment (anand-kumar-subramanian) [reply translib/translib.go:51]: Added more comments

File: translib/app_interface.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (renukamanavalan) [reply translib/app_interface.go:140]: So this function *always* succeeds. :-)

Curious: 
If so, either just return 'app' only or you may explicitly return Nil in the place of err. Why declare a variable, never use it, but return it

[the remark this answers is not in the document]
Comment (anand-kumar-subramanian) [reply translib/app_interface.go]: Made to return nil

**Pull request merged**